core/src/test/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractLibraryTestSuite.java

```diff
@@ -44,9 +53,16 @@ public void testInheritPermissionsException() {
     al.inheritPermissions(parent);
   }
 
-  // TODO - cs. This exception is never thrown! Remove it or add exception.
-  // @Test(expected = MalformedLibraryException.class)
-  // public final void testaddQc() {
-  //
-  // }
+  @Test
+  public final void testLibrarySerialization()  throws Exception {
+    Sample sample = new SampleImpl();
+    Library library = new LibraryImpl();
+    LibraryAdditionalInfo lai = new LibraryAdditionalInfoImpl();
+    library.setSample(sample);
+    lai.setTissueOrigin(new TissueOriginImpl());
+    lai.setTissueType(new TissueTypeImpl());
+    lai.setLibrary(library);
+    library.setLibraryAdditionalInfo(lai);
+    mapper.writer().writeValueAsString(library);
+  }
```

Comment:
Please add an assert to this test. (I guess, you know it's working when it doesn't throw an exception at the moment. But better to also positively test that the string contains a value that you want.) You can use the `containsString` matcher to avoid needing to match the entire string exactly.